mm2src/common/expirable_map.rs

```diff
@@ -0,0 +1,109 @@
+//! This module provides a cross-compatible map that associates values with keys and supports expiring entries.
+//!
+//! Designed for performance-oriented use-cases utilizing `FxHashMap` under the hood,
+//! and is not suitable for cryptographic purposes.
+
+use instant::Duration;
+use rustc_hash::FxHashMap;
+use std::hash::Hash;
+
+use crate::get_local_duration_since_epoch;
+
+#[derive(Clone, Debug)]
+struct ExpirableEntry<V> {
+    exp: Duration,
+    created_at: Duration,
+    value: V,
+}
```

Comment:
I already checked `time_cache`, but they were serving different purposes. I will try to merge them as much as possible (this will probably slightly change one of their behaviour).

Reply:
Done